Add no-stdlib build for mlkem-native (#1153)
pq-code-package/mlkem-native: examples/multilevel_build_no_stdlib/README.md

Author: Yes, thanks for the reminder!
I have addressed this and re-pushed the changes,
including `examples/basic/mlkem_native/mlkem/README.md` and `examples/README.md`,
and also added them to `base.yml` for the CI tests.

[//]: # (SPDX-License-Identifier: CC-BY-4.0)

# Multi-level build with no standard library

This directory contains a minimal example demonstrating how to build mlkem-native with support for all three security levels: ML-KEM-512, ML-KEM-768, and ML-KEM-1024, and so that level-independent code is shared. Unlike a simple [`multilevel_build`](../multilevel_build/README.md), this example also demonstrates how to replace `mlk_memcpy` and `mlk_memset` with custom implementations, allowing the build to proceed without using the standard library (stdlib).
In this example, only the C-backend of mlkem-native is used.

The library is built 3 times in different build directories `build/mlkem{512,768,1024}`. For the ML-KEM-512 build, we set
`MLK_CONFIG_MULTILEVEL_WITH_SHARED` to force the inclusion of all level-independent code in the
MLKEM512-build. For ML-KEM-768 and ML-KEM-1024, we set `MLK_CONFIG_MULTILEVEL_NO_SHARED` to not include any

> level-independent code, beside this, this example replace the `mlk_memcpy`, `mlk_memset` with the custom implementation memcpy and memset instead of using the stdlib by adding `example_no_stdlib_config.h`, and add the `-nostdlib` during objects file building,

Reviewer: ```suggestion
level-independent code, beside this, this example replace the `mlk_memcpy`, `mlk_memset` with the custom implementation memcpy and memset instead of using the stdlib by adding `example_no_stdlib_config.h`, and add `-nostdlib` to the compilation flags,
```

[//]: # (SPDX-License-Identifier: CC-BY-4.0)

# Multi-level build with no standard library

> This directory contains a minimal example for how to build mlkem-native with support for all 3 security levels

Reviewer: This description is copied 1:1 from the multilevel build? Can you please add a sentence about nostd lib?

> MLKEM-512, MLKEM-768, and MLKEM-1024, and so that level-independent code is shared. In this example, only the C-backend

Reviewer: The parameter sets are called ML-KEM-512 etc. Can you please change that *everywhere* in mlkem-native?
